packages/lit-html/src/lit-html.ts

```diff
@@ -226,6 +226,14 @@ export type TemplateResult<T extends ResultType = ResultType> = {
 };
 
 export type SVGTemplateResult = TemplateResult<typeof SVG_RESULT>;
+export type TemplateFactory = (options?: RenderOptions) => Template;
```

Comment:
It's sort of unfortunate that the factory means you have to call a function and get a memoized result for every TemplateResult. We also can't change this since it's technically public. Maybe consider not making this a factory and require the compiler to handle shady scoping?

Reply:
We have to do something to get options from the ChildPart into preparing the template. I can try to push this into core, but there will still be some check that the template was prepared.

Reply:
I got rid of the factory with a touch more logic in _commitTemplateResult. This makes compiled templates smaller.